### PR TITLE
Use local package dir and staging dir in solaris packager

### PR DESCRIPTION
**Pull request by sh9189**
Solaris packager does not use 'base_dir' variable in omnibus.rb. It hard codes '/var/cache/omnibus' directory while packaging. 
Health of solaris package fails due to solaris system libraries not being whilelisted.

This pull request fixes the above two issues. These were exposed while building omnibus packages for collectd (https://github.com/collectd/collectd)

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @sh9189

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by chef-supermarket** [comment]
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

**Comment by scotthain** [comment]
A number of these libraries are not part of a core Solaris installation (libfl, libcryptoutil, etc) and as such should not be on the global whitelist, in my opinion. They can be whitelisted in your core project as follows:
```
if solaris2?
  whitelist_file /libcryptoutil\.so/
end
```

The local cache dir issue is definitely a bug, thanks for catching that! Can you open a PR with just that change? Thank you!

**Comment by curiositycasualty** [comment]
Gunna have to +1 @scotthain thinking here. You're free to whitelist libs for your specific project, but the whitelist as it now exists needs only list libraries present in a base install of Solaris.

`base_dir` issue would be a great bugfix PR.

**Comment by sh9189** [comment]
@scotthain @curiositycasualty I updated the PR with the suggestions. Please take a look.

**Comment by sh9189** [comment]
@scotthain @curiositycasualty Can you please review?

**Comment by scotthain** [comment]
LGTM
@chef/omnibus-maintainers for extra review

File: lib/omnibus/packagers/solaris.rb [hunk 1]
```diff
@@ -35,11 +35,10 @@ class Packager::Solaris < Packager::Base
       copy_file("#{project.package_scripts_path}/postinst", '/tmp/pkgmk/postinstall')
       copy_file("#{project.package_scripts_path}/postrm", '/tmp/pkgmk/postremove')
 
-      Dir.chdir(staging_dir) do
```
Comment (sethvargo) [inline lib/omnibus/packagers/solaris.rb]: :+1: but I think we need a comment explaining why for the futures

Comment (schisamo) [inline lib/omnibus/packagers/solaris.rb]: Why was this `Dir.chdir` removed?

File: lib/omnibus/packagers/solaris.rb [hunk 2]
```diff
@@ -35,11 +35,10 @@ class Packager::Solaris < Packager::Base
...
-        shellout! "pkgmk -o -r #{install_dirname} -d /tmp/pkgmk -f /tmp/pkgmk/Prototype"
-        shellout! "pkgchk -vd /tmp/pkgmk #{project.package_name}"
-        shellout! "pkgtrans /tmp/pkgmk /var/cache/omnibus/pkg/#{package_name} #{project.package_name}"
-      end
+      shellout! "pkgmk -o -r #{install_dirname} -d /tmp/pkgmk -f /tmp/pkgmk/Prototype"
+      shellout! "pkgchk -vd /tmp/pkgmk #{project.package_name}"
+      abs_package_path = File.absolute_path(package_path) 
+      shellout! "pkgtrans /tmp/pkgmk #{abs_package_path} #{project.package_name}"
```
Comment (schisamo) [inline lib/omnibus/packagers/solaris.rb]: We shouldn't need to make this `File.absolute_path` call. If `package_path` needs to be expanded it should be done in `Omnibus::Packager::Base`: https://github.com/chef/omnibus/blob/c9937e6d3729aa72bf45c0bd77a6ff9d4c2c5d09/lib/omnibus/packagers/base.rb#L159-L168

So line 41 should just look like:

```ruby
shellout! "pkgtrans /tmp/pkgmk #{package_path} #{project.package_name}"
```

File: lib/omnibus/packagers/solaris.rb (hunk 1 above)
Comment (sh9189) [inline lib/omnibus/packagers/solaris.rb]: We don't need to chdir to staging_dir here. All commands pkgmk, pkgchk and pkgtrans take full paths as arguments.

**Comment by sh9189** [comment]
@schisamo I have updated the PR with your comments. Also solaris packager is not using staging_dir from base packager and is using /tmp/pkgmk as staging dir and this directory is not cleaned up at the end of packaging which leads to "leaks" as described in base packager code. I have made changes to use staging dir in solaris packager. @sethvargo @schisamo Please review.

File: lib/omnibus/packagers/solaris.rb [hunk 3]
```diff
@@ -58,6 +50,10 @@ def install_dirname
     def install_basename
       File.basename(project.install_dir)
     end
+    
+    def staging_dir_path(fileName)
+      File.join(staging_dir,fileName)
```
Comment (schisamo) [inline lib/omnibus/packagers/solaris.rb]: A few small Ruby style things:

* Don't camel case variable names. `fileName` should be `file_name`.
* Add a space after the comma: `staging_dir, file_name`.

**Comment by schisamo** [comment]
Modulo my style comment things look good @sh9189.

**Comment by sh9189** [comment]
@schisamo Thanks for the review. Made the suggested changes.

**Comment by schisamo** [comment]
I'm :+1:. Any additional review/comments @chef/omnibus-maintainers?

**Comment by yzl** [comment]
lgtm

**Comment by oferrigni** [comment]
Looks reasonable to me

**Comment by stevendanna** [comment]
:+1:

**Comment by curiositycasualty** [comment]
:+1:

**Pull request merged**